### ISSUE TITLE
[Question] PSNR score of LVSM is different in their own paper in RE10K

### ISSUE THREAD

**Issue by Kin-Zhang** [hwjiang1510/RayZer#4]
Thanks for sharing work!

I'm curious about the score reported different in your baseline LVSM.

![Image](https://github.com/user-attachments/assets/67d6cccb-9086-4919-956b-ce739645ad73)

I downloaded LVSM pretrained weight and inference testing through their repo guideline and the score aligned with their reported 29.67. However, rayzer paper looks have different score and lower. One reason mybe highlight in the paper that camera annotation come from COLMAP. While PixelSplat also **used sfm software** to estimate camera pose. I assume it's similar to rayzer did:
![Image](https://github.com/user-attachments/assets/dd86bc86-a419-4e4c-9f17-24e32129ddb2)

Could I know more detail about why there is a large difference between rayzer table (colmap) and lvsm table (sft software fromp pixelsplat). 

Related issue: [pose discussion](https://github.com/hwjiang1510/RayZer/issues/1)

**Comment by hwjiang1510** [comment]
Hi Qingwen,

I don't think this is related to COLMAP, because COLMAP should be the SfM tool used in pixelsplat.

I think the reason is that we use different settings for training LVSM baseline in RayZer paper compared with the one in LVSM original paper.

First, the baseline in RayZer is the encoder-decoder LVSM, rather than the decoder-only LVSM, as mentioned in the baseline section. The decoder-only LVSM gives high-quality novel views, but the computation cost is very high (during inference). Thus, we use the encoder-decoder one, and it's naturally a bit worse than the one you highlighted in the table from LVSM paper.

Second, we use a slightly smaller number of parameter for the LVSM baseline in RayZer. As mentioned in the baseline section, we use 16 transformer layers for LVSM, which is smaller than the one using 24 layers in LVSM paper. And using smaller model will naturally leads to lower performance.

Third, we recently tried train LVSM and RayZer with 24 transformer layers (note that RayZer has 8 additionally layers for pose estimation). On DL3DV, they show PSNR of 24.85 (LVSM) and 25.36 (RayZer), with about 1 dB PSNR gain from the 16 layer ones. Both results are for the uniform sampling setting. This 1 dB gap is also (roughly) the gap on RealEstate10k between original LVSM and our reported LVSM number in RayZer paper.

Finally, the number of the two LVSM model are not directly comparable, as we have different setting of input images, view sampling range, etc. We have ensured the LVSM results reported in RayZer paper have the same settings with RayZer, but they are different from LVSM paper.

**Comment by Kin-Zhang** [comment]
Thanks for the explanation ❤️. I mostly got your point here. (*speed, *model size and *different evaluation setting) Sorry for the many questions here, trying to understand better.

1. [speed] During the step of decreasing the layer number in LVSM already, with half the layer number of the original decoder version, will the computation cost still be very high? And maybe it would be good to point out somehow in the paper about the speed advantage of Rayzer if the speed is the reason we choose the encoder-decoder version?
2. [size] I saw in LVSM tab. 2, they have the ablation study on model size, even with only a 6-layer decoder-only works better on RE10K (27.62) than rayzer reported. (6-layer model param only 43M, which is 3x smaller than the paper compared with) .
3. [rayzer] If I understand correctly, the difference between lvsm encoder-decoder version and rayzer is mainly the camera estimation added as Fig. 3 pink part? Then that makes more sense to compare lvsm encoder-decoder version as one of the important ablation studies. (Please point out if I'm wrong here, trying to find the difference in rayzer's latent scene reconstruction and LVSM's encoder)
4. Accordingly to the 3rd point, I'm curious whether Rayzer should directly use camera estimation and **decoder-only**, does that work even better?
5. [evaluation] As another point on `setting of input images, view sampling range`, thanks for reminding me on this part, as I found in rayzer the number of input and target are actually increased rather than follow pixelsplat&LVSM? Do we have any results following the same setting as previous works? (Is that because a few inputs could be hard for Rayzer because of the pose estimation?

Thanks again for the previous quick answer!

**Comment by hwjiang1510** [comment]
1. Decoder-only LVSM is naturally computational expensive, because you have to run the entire model to render a novel view. Reducing the number of layer helps, but it cannot solve the problem. For speed advantage, we actually don't care about that, as they are detailed model designs, while our goal is to verify the possibility of self-supervised learning. Using encoder-decoder is the choice to accelerate our experiment iteration during development.

2. Exactly, the Encoder in LVSM will lead to worse performance. For example, 3 layer encoder + 6 layer decoder will be worse than 6 layer decoder-only, even though we have more parameters for the encoder-decoder one in this case. Encoder helps us speed up, but any encoder (compression of the input images) leads to performance drop.

3. Exactly, RayZer has three stages as illustrated in its figure 3, where the last two stages is basically an encoder-decoder LVSM.

4. We can use decoder-only and this will lead to better performance. But I don't think decoder-only is the correct choice beyond the novel view synthesis task. We need more 3D awareness of latent representation.

5. We haven't try for example two input views on RealEstate. I think that will also work, and I do expect the performance will be similar, as RealEstate is a very simple dataset and not that meaningful anymore. The results on DL3DV is following the similar settings from previous works, e.g., MegaSynth.

**Comment by Kin-Zhang** [comment]
Thanks again! ❤️ Appreciate your answer!

> We can use decoder-only, and this will lead to better performance. 

Is this a conclusion that you already tried internally, or is it more of a guess? You also mention here that decoder-only is not always the best choice, and according to adding camera estimation, maybe it won't have better results as there is less 3D awareness.

Other question:
6. I saw the training details on `32 A100 GPUs`, could you also provide the training time?
7. I'm curious more about training, as the pose is SSL. How was the convergence plot with only reconstruction as loss? As I knew, sometimes without GT supervision, it will diverge easily on pose estimation.


Thanks for your reply! Learn a lot.
